### PR TITLE
Add ability to show popups (and popup menus) at a relative location

### PR DESCRIPTION
**Pull request by andydotxyz**
### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Public APIs match existing style and have Since: line.

### CODE REVIEW

File: widget/popup.go [hunk 1]
```diff
@@ -69,6 +69,15 @@ func (p *PopUp) ShowAtPosition(pos fyne.Position) {
 	p.Show()
 }
 
+// ShowAtRelativePosition shows this pop-up at the given position relative to stated object.
+//
+// Since 2.4
+func (p *PopUp) ShowAtRelativePosition(rel fyne.Position, to fyne.CanvasObject) {
+	withRelativePosition(rel, to, func(pos fyne.Position) {
+		p.ShowAtPosition(pos)
+	})
```
Comment (Jacalz) [inline widget/popup.go:78]: You should be able to drop the wrapper function here.
```suggestion
	withRelativePosition(rel, to, p.ShowAtPosition)
```

File: widget/popup_menu.go [hunk 2]
```diff
@@ -95,6 +105,15 @@ func (p *PopUpMenu) ShowAtPosition(pos fyne.Position) {
 	p.Show()
 }
 
+// ShowAtRelativePosition shows the pop-up menu at the position relative to given object.
+//
+// Since 2.4
+func (p *PopUpMenu) ShowAtRelativePosition(rel fyne.Position, to fyne.CanvasObject) {
+	withRelativePosition(rel, to, func(pos fyne.Position) {
+		p.ShowAtPosition(pos)
+	})
```
Comment (Jacalz) [inline widget/popup_menu.go:114]: The same simplification can be applied here.
```suggestion
	withRelativePosition(rel, to, p.ShowAtPosition)
```

**Review by Jacalz — CHANGES_REQUESTED** [review]
This seems like a nice and useful addition. Thanks. I just left two comments inline about removing two unnecessary anonymous wrapper functions.

**Comment by andydotxyz** [comment]
Great suggestions, added

**Review by Jacalz — APPROVED** [review]

**Pull request merged**